qemu/tests/test_control_virtqueue.py

```diff
@@ -0,0 +1,51 @@
+from virttest import error_context
+from virttest import utils_net
+from virttest.utils_test.qemu import MemoryHotplugTest
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test vDPA control virtqueue
+
+    1) Boot a guest and qemu must have space for more memory slots
+    2) Change mac address 2^16-1 times
+    3) Hotplug 1G memory to update device
+    4) Try to update the mac again and it should mac address can be changed
+       normally
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = int(params.get("login_timeout", 360))
+    session_serial = vm.wait_for_serial_login(timeout=timeout)
+    size_mem = params.get("size_mem")
+    change_cmd = params["change_cmd"]
+    target_mem = params["target_mem"]
+    old_mac = vm.get_mac_address(0)
+    new_mac = vm.virtnet.generate_mac_address(0)
+    interface = utils_net.get_linux_ifname(session_serial, old_mac)
+    test.log.info("change mac address 2^16-1 times")
+    change_times = params.get_numeric("change_times")
+    for i in range(change_times):
+        output = session_serial.cmd_output_safe(change_cmd % (new_mac, interface))
```

Comment:
Here the new MAC always `new_mac = vm.virtnet.generate_mac_address(0)`, should we move it to the for loop?

Reply:
> Here the new MAC always `new_mac = vm.virtnet.generate_mac_address(0)`, should we move it to the for loop?

Hi @PaulYuuu Good question, I confirmed this question with developer in the past, the developer told me we just used this method to test changes through CVQ, so it can always used a same mac address expect original mac.